compiler/plugins/target/ROCM/builtins/ukernel/BUILD.bazel

```diff
@@ -0,0 +1,61 @@
+# Copyright 2024 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+load("//build_tools/bazel:build_defs.oss.bzl", "iree_cmake_extra_content", "iree_runtime_cc_library")
+load("//build_tools/bazel:iree_bitcode_library.bzl", "iree_amdgpu_bitcode_library")
+load("//build_tools/embed_data:build_defs.bzl", "iree_c_embed_data")
+
+package(
+    default_visibility = ["//visibility:public"],
+    features = ["layering_check"],
+    licenses = ["notice"],  # Apache 2.0
+)
+
+iree_cmake_extra_content(
+    content = """
+if(NOT IREE_TARGET_BACKEND_ROCM)
+  return()
+endif()
+""",
+    inline = True,
+)
+
+gpu_archs = [
+    "gfx90a",
+    "gfx942",
+    "gfx1030",
+    "gfx1100",
+]
```

Comment:
How many of these architectures are we going to need [built into the compiler]? How much binary size and compile time does each add?

If less than 10, that seems acceptable. Any more than that and I'd worry about maintenance, binary size, and compile time costs.

Some suggestions:

* Could add a comment here with a link to a list of available architectures and some description of which are worth including here. Docs: https://iree.dev/guides/deployment-configurations/gpu-rocm/#compile-a-program and https://llvm.org/docs/AMDGPUUsage.html#processors could work.
* Would we want to add a CMake option to add other architectures to the list? (and possibly Bazel...)
* Can `iree-compile` generate these on-demand if a user passes a target chip that isn't in the list of packaged ukernels? (Or, if these are embedded in the runtime, can they be configured when building programs like `iree-run-module`?)

Reply:
The current `argmax` ukernel being added here is more aggressive in supporting "all" architectures than what I envision adding going forward.

I mostly envision ukernels for `iree_gpu.multi_mma` , each painstakingly hand-written in amdgpu assembly (wrapped as inline-assembly into a C form). My whole motivation is to work around defects in the LLVM AMDGPU backend, and assembly is the only thing that doesn't ultimately depend on that. So, the ukernels that I will be adding will by construction be specific to one architecture, and expensive in engineering time.

This argmax ukernel is different: written in C and compiled for "all currently relevant architectures", what it works around is the difficulty of generating efficient code for this kind of kernel in the MLIR programming model, due to the very specific things that argmax needs to do (see wave reductions (`wfred`) and ballots in the C sources).

Even then, even for such `argmax`-style C ukernels, the number of architectures is still less than 10 in any foreseeable future -- since this is an optimization-only thing (argmax still works without ukernels, which is also why they are not enabled by default) --- we never need to support more architectures than we really care about performance on. At the moment, the number of architectures that we really care about performance on, equals 1 (that is: CDNA3)...

Reply:
Thanks, that helps a lot. I don't have any extra concerns with this approach for now then.

Can we get some of that text into a comment somewhere in the source and/or a markdown file?

Reply:
I just added an allusion to that in addressing your review comment on BUILD.bazel.  I expect that the code is going to start conveying intent better once (1) the next PR moves the code loading the bitcode to LowerToUkernels (see other reply) and (2) I start adding multi_mma ukernels that provide the better model for the generic case going forward. At that point we'll have fewer things requiring comment (because always best when the code speaks for itself) and more stable places to add comments on what still needs commenting.